src/api/app/models/user.rb

```diff
@@ -127,6 +128,18 @@ def create_home_project
     self.login_failure_count = 0 if login_failure_count.nil?
   end
 
+  # autocomplete method
+  # return either an array of users or an array of hashes
+  def self.autocomplete(prefix = nil, is_token = false)
```

Comment:
I would prefer to have a separate method for autocomplete tokens, that way you don't have to check it for every single item neither to pass a param for it and the collect would be super easy in each case. At the same time you can use each method in the corresponding action of the controller.

Reply:
I also don't like to check the is_token in the loop as it won't change. Also in case of is_token = false you iterate over the items instead of just returning them.
What about:

```
result =  where('login LIKE ?', "#{prefix}%").pluck(:login)
is_token ? result.collect { |user| { name: user } }: result
```